test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -4576,6 +4576,61 @@ BOOST_AUTO_TEST_CASE(constructing_enums_from_ints)
 	ABI_CHECK(callContractFunction("test()"), encodeArgs(1));
 }
 
+BOOST_AUTO_TEST_CASE(struct_referencing)
+{
+	static char const* sourceCode = R"(
+		pragma experimental ABIEncoderV2;
+		interface I {
+			struct S { int a; }
+		}
+		library L {
+			struct S { int a; }
+			function f() public pure returns (S) {
+				S memory s;
+				s.a = 3;
+				return s;
+			}
+			function g() public pure returns (I.S) {
+				I.S memory s;
+				s.a = 4;
+				return s;
+			}
```

Comment:
Please add two functions to the library to see if the selectors work as expected:

```
function a(S storage) public pure returns (uint) { return 1; }
function a(I.S storage) public pure returns (uint) { return 2; }
```

and then define two variables of the respective struct type in `C` and then call the library on them.

Reply:
done.